File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/MessageSender.java
package uk.gov.hmcts.reform.bulkscan.orchestrator.helper;

import com.microsoft.azure.servicebus.IMessageHandler;
import com.microsoft.azure.servicebus.Message;

public class MessageSender {

    private final IMessageHandler processor;

    public MessageSender(IMessageHandler processor) {
        this.processor = processor;
    }

    public void send(Message message) {
        processor.onMessageAsync(message).join();
    }
}

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/MessageSender.kt
package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers

import com.microsoft.azure.servicebus.IMessageHandler
import com.microsoft.azure.servicebus.Message
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired

class MessageSender(@Autowired private val processor: IMessageHandler) {
    val logger: Logger = LoggerFactory.getLogger(this::class.simpleName)
    fun send(message: Message) {
        processor.onMessageAsync(message).get()
    }
}
